app/controllers/users/notification_settings_controller.rb

```diff
@@ -0,0 +1,61 @@
+module Users
+  class NotificationSettingsController < ApplicationController
+    before_action :raise_suspended
+    before_action :authenticate_user!
+    after_action :verify_authorized
+
+    ALLOWED_PARAMS = %i[email_badge_notifications
+                        email_comment_notifications
+                        email_community_mod_newsletter
+                        email_connect_messages
+                        email_digest_periodic
+                        email_follower_notifications
+                        email_membership_newsletter
+                        email_mention_notifications
+                        email_newsletter
+                        email_tag_mod_newsletter
+                        email_unread_notifications
+                        mobile_comment_notifications
+                        mod_roundrobin_notifications
+                        reaction_notifications
+                        welcome_notifications].freeze
+
+    def update
+      authorize current_user, policy_class: UserPolicy
+
+      if current_user.notification_setting.update(users_notification_setting_params)
+        flash[:settings_notice] = "Your notification settings have been updated."
+      else
+        Honeycomb.add_field("error", current_user.notification_setting.errors.messages.compact_blank)
+        Honeycomb.add_field("errored", true)
+        flash[:error] = current_user.notification_setting.errors_as_sentence
+      end
+      redirect_to user_settings_path(:notifications)
+    end
+
+    def onboarding_notifications_checkbox_update
+      if params[:notifications]
+        permitted_params = %i[email_newsletter email_digest_periodic]
+        current_user.notification_setting.assign_attributes(params[:notifications].permit(permitted_params))
+      end
+
+      current_user.saw_onboarding = true
+      authorize User
+      render_update_response(current_user.notification_setting.save)
+    end
+
+    private
+
+    def render_update_response(success, errors = nil)
+      status = success ? 200 : 422
+
+      respond_to do |format|
+        format.json { render json: { errors: errors }, status: status }
```

Comment:
This should actually return the proper response status code since the status is outside of the `:json` hash which only contains the error. I was able to write and validate this with a test so I think we are good to go here.

Reply:
Oops, sorry, I totally misread this line, thanks for verifying!